transport/tlscommon/tls.go

```diff
@@ -66,7 +66,14 @@ func LoadCertificate(config *CertificateConfig) (*tls.Certificate, error) {
 		return nil, err
 	}
 
-	log.Debugf("Loading certificate: %v and key %v", certificate, key)
+	// Do not log the key if it was provided as a string in the configuration to avoid
+	// leaking private keys in the debug logs. Log when the key when it is a file path.
+	if IsPEMString(key) {
+		log.Debugf("Loading certificate: %v with key from PEM string in config", certificate)
+	} else {
+		log.Debugf("Loading certificate: %v and key %v", certificate, key)
```

Comment:
I think the key snuck back in the debug log.

Reply:
This should only log the key if `IsPEMString(key)` is false, which essentially detects whether the key contents are a string in the config and allows logging the path we are using when loading it from a file.

I can remove this entirely if we are concerned about it.

Reply:
🤦 ok, so it should be a file path at that point which is fine.